haskell/toolchain.bzl

```
@@ -80,7 +80,11 @@ while IFS= read -r line; do compile_flags+=("$line"); done < %s
 while IFS= read -r line; do extra_args+=("$line"); done < %s
 while IFS= read -r line; do param_file_args+=("$line"); done < %s
 
-"${compile_flags[@]}" "${extra_args[@]}" ${param_file_args+"${param_file_args[@]}"}
+# XXX Workaround https://gitlab.haskell.org/ghc/ghc/merge_requests/1308.
+output=$("${compile_flags[@]}" "${extra_args[@]}" ${param_file_args+"${param_file_args[@]}"} 2>&1)
```

I think we can avoid storing the output in a variable and instead stream it right away.

```suggestion
set -o pipefail
"${compile_flags[@]}" "${extra_args[@]}" ${param_file_args+"${param_file_args[@]}"} 2>&1 \
  | while IFS= read -r line; do [[ $line =~ ^Loaded ]] || echo "$line"; done >&2
```

`set -o pipefail` forwards the exit code if GHC fails, so the `rc=$?` below becomes unnecessary.